### PR TITLE
[Create Environment] Add role to EKS cluster for post-provisioning access

### PR DESCRIPTION
**Pull request by gurevichdmitry**
### Summary of your changes

This PR addresses the need to grant the `Developer_eks` role access to the AWS EKS cluster after it has been provisioned. The `Developer_eks` role is essential for developers to interact with the cluster.

### Changes Made

- **Added Developer_eks Role**: Introduced the `Developer_eks` IAM role, which is essential for developer access to the EKS cluster.
- **Updated Create Environment Workflow**: Modified the `create environment` workflow to use a single command for destroying the environment.
- **Updated delete_env script**: Enhanced the delete_env script to support the correct destruction of both clusters created before and after the introduction of the `Developer_eks` role. 
- **Documentation Updates**: Updated the documentation to provide clear and detailed instructions on how developers can connect to the EKS cluster using the `Developer_eks` role.

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: deploy/cloud/README.md [hunk 1]
```diff
@@ -79,6 +79,17 @@ See the ssh command `terraform output -raw cloudbeat_ssh_cmd`
 **Delete environment:**
 
 ```bash
+#!/bin/bash
+set -e
+
+RESOURCE_NAME="module.eks.module.eks.kubernetes_config_map_v1_data.aws_auth"
```
Comment (oren-zohar) [inline deploy/cloud/README.md:85]: Can you explain the reason behind the name `module.eks.module.eks`?

Comment (gurevichdmitry) [reply deploy/cloud/README.md:85]: The `test-environment` terraform uses modules from the `deploy/cloud` project.

`module.eks`: This is the main module containing the `eks` module used in `test-environment`.

`module.eks.module.eks`: It's an inner module within the main module, defining EKS-related settings in the `deploy/cloud` project.

The repetition of `eks` in both names can make it confusing.

Comment (gurevichdmitry) [reply deploy/cloud/README.md]: refactored

**Review by oren-zohar — APPROVED** [review]
LGTM

**Pull request merged**